### DISCUSSION TITLE
Auto-generate nav in sidebar?

### DISCUSSION THREAD

**Discussion by boozedog** [vuejs/vitepress#2330] [Q&A]
Using the default theme, Is it possible to auto-generate the sidebar nav? Rather than having to add manually to `config.ts` every time I add a new `.md` in my `docs` folder?

**Comment by brc-dd** [comment]
Duplicate of #1297

**Comment by boozedog** [comment]
Thanks @brc-dd ! However, I am not able to understand #1297 , can you tell me what was the solution here? I do see someone put some code in there

**Reply by brc-dd** [reply]
The basic idea is to write some custom script in your config file that traverses your markdown pages and generates the sidebar for you. There are some [community plugins](https://www.npmjs.com/search?q=vitepress%20sidebar) too which you can use or refer for implementation.

**Comment by boozedog** [comment]
For others who may be wondering the same thing, I found a plugin that seems to work well here: https://github.com/jooy2/vitepress-sidebar
